### PR TITLE
Added end point to get Compliance Data Of a Device

### PR DESCRIPTION
**Pull request by madhawap**

### CODE REVIEW

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java [hunk 1]
```diff
@@ -301,4 +304,48 @@ public Response getEffectivePolicyOfDevice(@PathParam("type") String type,
...
+                    "' device, which carries the id '" + id + "'";
+            log.error(msg, e);
+            return Response.serverError().entity(
+                    new ErrorResponse.ErrorResponseBuilder().setMessage(msg).build()).build();
+        }
+
+        if (policy == null) {
+            deviceCompliance.setDeviceID(Integer.valueOf(id));
+            deviceCompliance.setComplianceData(null);
+            //deviceCompliance.setCode(0001l); //code 0001 means no compliance data related to the device
```
Comment (madawas) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java:331]: @madhawap Can we remove this commented out line?

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java [hunk 2]
```diff
@@ -301,4 +304,48 @@ public Response getEffectivePolicyOfDevice(@PathParam("type") String type,
...
+            return Response.status(Response.Status.OK).entity(deviceCompliance).build();
+        } else {
+            try {
+                policyManagementService = DeviceMgtAPIUtils.getPolicyManagementService();
+                complianceData = policyManagementService.getDeviceCompliance(
+                        new DeviceIdentifier(id, type));
+                deviceCompliance.setDeviceID(Integer.valueOf(id));
+                deviceCompliance.setComplianceData(complianceData);
+                //deviceCompliance.setCode(0002l); //code 0002 means there are compliance data related to the device
```
Comment (madawas) [inline components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java:340]: @madhawap Can we remove this commented out line?

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java (hunk 1 above)
Comment (madhawap) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java:331]: Yes. Thanks for pointing out. Corrected with following PR.
https://github.com/wso2/carbon-device-mgt/pull/734

File: components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java (hunk 2 above)
Comment (madhawap) [reply components/device-mgt/org.wso2.carbon.device.mgt.api/src/main/java/org/wso2/carbon/device/mgt/jaxrs/service/impl/DeviceManagementServiceImpl.java:340]: Corrected with https://github.com/wso2/carbon-device-mgt/pull/734

**Pull request merged**